File: chap3/CalculFormel.java
// Ces programmes sont sous licence CeCILL-B V1.
class CalculFormel {

  public static void main (String [] args) {
    int i;
    double x,y,c;
    double [] t,u; 
    t = new double [4];   
    u = new double [4];   
    t[3] = 2;
    t[2] = 8;
    t[1] = 7;
    t[0] = 3;

    //Affichage de la fonction
    for (i = 0; i <= 3; i = i + 1) {
      System.out.print(t[i]);
      System.out.print(" ");
      if (i != 0) {
        System.out.print("x");
        if (i != 1) {
          System.out.print("^");
          System.out.print(i);}}
      if (i != 3) {
        System.out.print(" + ");}}
    System.out.println();


    //Calcul et affichage de sa valeur
    x = 5;
    y = 0;
    c = 1;
    for (i = 0; i <= 3; i = i + 1) {
      y = y + t[i] * c;
      c = c * x;}
    System.out.println(y);

    //Calcul de sa dérivée
    for (i = 0; i <= 2; i = i + 1) {
      u[i] = t[i+1] * (i + 1);}
    u[3] = 0;

    //Affichage de sa dérivée
    for (i = 0; i <= 3; i = i + 1) {
      System.out.print(u[i]);
      System.out.print(" ");
      if (i != 0) {
        System.out.print("x");
        if (i != 1) {
          System.out.print("^");
          System.out.print(i);}}
      if (i != 3) {
        System.out.print(" + ");}}
    System.out.println();}}

File: chap3/CalculFormel.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>

using namespace std;

const int DEGRE_MAX = 3;

int main()
{
   int i;
   double x,y,c;
   double t[DEGRE_MAX + 1];
   double u[DEGRE_MAX + 1]; 

   t[3] = 2;
   t[2] = 8;
   t[1] = 7;
   t[0] = 3;

   //Affichage de la fonction
   for (i = 0; i <= DEGRE_MAX; i = i + 1)
   {
      cout << t[i];
      if (i != 0) 
      {
         cout << " x";
         if (i != 1)
         {
            cout << "^";
            cout << i;
         }
      }
      if (i != DEGRE_MAX)
      {
         cout << " + ";
      }
   }
   cout << endl;


   //Calcul et affichage de sa valeur
   x = 5;
   y = 0;
   c = 1;
   for (i = 0; i <= DEGRE_MAX; i = i + 1) 
   {
      y = y + t[i] * c;
      c = c * x;
   }
   cout << y << endl;

   //Calcul de sa dérivée
   for (i = 0; i <= DEGRE_MAX - 1; i = i + 1)
   {
      u[i] = t[i+1] * (i + 1);
   }
   u[DEGRE_MAX] = 0;

   //Affichage de sa dérivée
   for (i = 0; i <= DEGRE_MAX; i = i + 1)
   {
      cout << u[i];
      if (i != 0) 
      {
         cout << " x";
         if (i != 1)
         {
            cout << "^";
            cout << i;
         }
      }
      if (i != DEGRE_MAX) 
      {
         cout << " + ";
      }
   }
   cout << endl;
}
